modules/terraform.nix

```diff
@@ -5,6 +5,8 @@ let
 
   kms2region = kms: if kms == null then null else builtins.elemAt (lib.splitString ":" kms) 3;
 
+  relEncryptedFolder = lib.last (builtins.split "-" (toString config.secrets.encryptedRoot));
```

Comment:
Suggest making relEncryptedFolder prem infraType friendly here also.  This older def breaks the prem case due to both the split match and the config secrets reference.  Example [here](https://github.com/input-output-hk/bitte/blob/master/modules/terraform/tf-options.nix#L34-L39).

Reply:
I see this logic repeated all over the place. I think it makes more sense to move your version into a single source of truth and use that throughout the codebase. But that would make more sense in its own PR (that I would be happy to create).

Reply:
Here is another example to make it clearer: [link](https://github.com/input-output-hk/bitte/pull/105/files#diff-1582854d2aa689ad355e6623148d243c202547dfeb78c8a708621314e427a6dbR9-R11).  This logic diff which is scattered should be disappearing in the future as the prem and aws encryption re-converge on the same module usage: sops instead of age for prem case most likely.  So it's up to you whether you'd prefer to do a faster couple line change now to maintain compat, or invest more time into cleaning something up that will be disappearing in the near future anyway.